frontend/javascripts/oxalis/model/bucket_data_handling/data_cube.ts

```diff
@@ -272,40 +332,48 @@ class DataCube {
   // NULL_BUCKET if the bucket cannot possibly exist, e.g. because it is
   // outside the dataset's bounding box.
   getOrCreateBucket(address: BucketAddress): Bucket {
-    if (!this.isWithinBounds(address)) {
-      return this.getNullBucket();
-    }
-
-    let bucket = this.getBucket(address, true);
+    let bucket = this.getBucket(address);
 
     if (bucket instanceof NullBucket) {
-      bucket = this.createBucket(address);
+      const containment = this.checkContainment(address);
+      if (containment.type === "no") {
+        return this.getNullBucket();
+      }
+      bucket = this.createBucket(address, containment);
     }
 
     return bucket;
   }
 
   // Returns the Bucket object if it exists, or NULL_BUCKET otherwise.
-  getBucket(address: BucketAddress, skipBoundsCheck: boolean = false): Bucket {
-    if (!skipBoundsCheck && !this.isWithinBounds(address)) {
-      return this.getNullBucket();
-    }
-
+  getBucket(address: BucketAddress): Bucket {
     const [bucketIndex, cube] = this.getBucketIndexAndCube(address);
 
     if (bucketIndex != null && cube != null) {
       const bucket = cube.data.get(bucketIndex);
 
-      if (bucket != null) {
+      // We double-check that the address of the bucket matches the requested
+      // address. If the address is outside of the layer's bbox, the linearization
+      // of the address into one index might collide with another bucket (that is
+      // within the bbox) which is why the check is necessary.
+      // We use slice to ignore the additional coordinates (this is mostly done
+      // to ignore annoying cases like null vs [] which have identical semantics).
+      if (bucket != null && _.isEqual(address.slice(0, 4), bucket.zoomedAddress.slice(0, 4))) {
```

Comment:
As you previously cared for preformance, `V4.isEqual` might be faster:

```suggestion
      if (bucket != null && V4.isEquals(address.slice(0, 4), bucket.zoomedAddress.slice(0, 4))) {
```
Note: Maybe casting is needed here for TS to assume the correct type and not complain.